### PR TITLE
Initial work on import map endpoint

### PR DESCRIPTION
**Pull request by joeldenning**
This is the initial work on the import map endpoints. The idea is that each organization that buys single-spa foundry will have an `orgKey` and will be able to download one or more import maps from our public CDN. The import maps are stored in Cloudflare KV (key value), which acts as a database but is a lot cheaper and faster. Our "customer app" for single-spa foundry (for our customers) will put values into Cloudflare KV when the users buy the service and make customizations and such. The "end user app" (app for our customers' customers) will read the values from KV when handling responses. The end user app needs to find the correct balance between defensive programming (can't trust the customer app to never have bugs in it when inserting into the KV) and properly surfacing errors when things are unrecoverable.

I've attached some screenshots below showing what responses look like when data is in the KV vs when it is not:

image

image

### CODE REVIEW

File: src/handleImportMap.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  } catch (err) {
+    // Defensive programming so that we don't take down their production applications
+    // if KV has invalid JSON in it or is down
+    console.error(err);
+  }
+
+  let importMap: ImportMap;
+
+  if (importMapKV) {
+    importMap = importMapKV;
```
Comment (Qirhi) [inline src/handleImportMap.ts:50]: Is Line 50 written because importMap does not allow null and importMapKV does?

File: src/main.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+addEventListener("fetch", (evt: FetchEvent) => {
+  evt.respondWith(handleRequest(evt.request));
+});
+
+const routeHandlers: RouteHandlers = {
+  "/:orgKey/:importMapName.importmap": handleImportMap,
+};
+
+const routeMatchers: RouteMatchers = Object.entries(routeHandlers).map(
+  ([path, handler]) => [match(path), handler]
```
Comment (Qirhi) [inline src/main.ts:14]: Where does the array come from on either side of the fat arrow?

File: src/main.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+);
+
+async function handleRequest(request: Request) {
+  const requestUrl = new URL(request.url);
+
+  let routeHandler: RouteHandler | undefined,
+    matchResult: MatchResult | false = false;
```
Comment (Qirhi) [inline src/main.ts:21]: Could this also be a `let`?

File: src/main.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  // Find which route handler to call for this request
+  // We don't have express to do this automatically for us, but are
```
Comment (Qirhi) [inline src/main.ts:24]: I'm trying to understand why we don't have express again.  Where does this code live?

**Review by Qirhi — APPROVED** [review]
At a high level I understand the code in this PR and what it is doing.  Nice work!

File: src/handleImportMap.ts (hunk 1 above)
Comment (joeldenning) [reply src/handleImportMap.ts:50]: yes

File: src/main.ts (hunk 2 above)
Comment (joeldenning) [reply src/main.ts:14]: Object entries returns an array of arrays. Each inner array has two items - the object key and object value. So the array on the left side is coming from Object.entries. Read more at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

The array on the right side is one that my code is creating. We're inside of a `.map()`, so whatever we return will be the new value in the newly created array. The code here is choosing to return an array where the first item is a route matcher function and second item is the route handler.

File: src/main.ts (hunk 3 above)
Comment (joeldenning) [reply src/main.ts:21]: It is. You can declare multiple variables at once by doing `let var1, var2`. Both var1 and var2 in that example are `let` variables. In the case of the code here, both `routeHandler` and `matchResult` are `let` variables, declared in the same statement and separated by a comma on line 20

File: src/main.ts (hunk 4 above)
Comment (joeldenning) [reply src/main.ts:24]: It's inside of cloudflare workers. See https://developers.cloudflare.com/workers/

**Review by Qirhi — APPROVED** [review]
This looks great!

**Pull request merged**